app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt

```diff
@@ -256,19 +266,62 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
-    // TODO draw lines
-    private fun drawLinesDetection(roadElements: List<RoadDescriptionMapper.RoadElement>) {
-        lines_detections_container.removeAllViews()
+    private fun drawLinesDetection(roadDescription: RoadDescription) {
 
-        for (roadElement in roadElements) {
+        fun getImageView(isFirst:Boolean = false): ImageView {
             val image = ImageView(this)
-            val lp = ViewGroup.MarginLayoutParams(lineSize, ViewGroup.LayoutParams.WRAP_CONTENT)
+            val lp = ViewGroup.MarginLayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT,lineHeight)
+            if(isFirst) {
+                lp.marginStart = margin
+            }
+            lp.marginEnd = margin
             image.layoutParams = lp
-            image.paddingLeft
+            return image
+        }
 
-            image.setImageResource(roadElement.drawableResourceId)
+        fun MarkingType.imageSource(isForward: Boolean = false): Int {
```

Comment:
nit: can just write `fun MarkingType.imageSource(isForward: Boolean = false): = when (this) ...`